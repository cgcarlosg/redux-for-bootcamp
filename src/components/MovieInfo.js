import React from "react";
import { imgPost } from "../constants";
import Styles from "../assets/stylesheet/MovieInfo.module.css";
import stars from "../assets/images/stars.svg"

const MovieInfo = ({ movie }) => {
  return (
    <div>
      <h1 className={Styles.title}>{movie.title}</h1>
      <div
        className={Styles.poster}
        style={{
          backgroundImage: `url(${imgPost + movie.backdrop_path})`,
        }}
      ></div>
      <div className={Styles.info}>
        <div className={Styles.text}>
          <h2> Description </h2>
          <p>{movie.overview}</p>
          <p>{movie.tagline}</p>
        </div>
        <div className={Styles.rates}>
            <img className={Styles.star} src={stars} alt="Star"/>
            <h1>{movie.vote_average}</h1>
        </div>
      </div>
    </div>
  );
};

export default MovieInfo;