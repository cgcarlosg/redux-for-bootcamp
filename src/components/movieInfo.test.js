import React from 'react';
import renderer from 'react-test-renderer';
import MovieInfo from './MovieInfo';


const movie = {
        "id": 19404,
        "original_language": "us",
        "original_title": "original",
        "overview": "Raj is a rich",
        "popularity": 19.252,
        "poster_path": "/2CAL2433ZeIihfX1Hb2139CX0pW.jpg",
        "release_date": "1995-10-20",
        "title": "Dilwale Dulhania Le Jayenge",
        "video": false,
        "vote_average": 8.7,
        "vote_count": 2996
};

it('Renders MovieInfo', () => {
  const popular = renderer
    .create(
      <MovieInfo movie={movie} />,
    )
    .toJSON();

  expect(popular).toMatchSnapshot();
});


