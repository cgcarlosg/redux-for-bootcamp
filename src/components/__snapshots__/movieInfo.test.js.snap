// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders MovieInfo 1`] = `
<div>
  <h1
    className="title"
  >
    Dilwale Dulhania Le Jayenge
  </h1>
  <div
    className="poster"
    style={
      Object {
        "backgroundImage": "url(https://image.tmdb.org/t/p/w1280undefined)",
      }
    }
  />
  <div
    className="info"
  >
    <div
      className="text"
    >
      <h2>
         Description 
      </h2>
      <p>
        Raj is a rich
      </p>
      <p />
    </div>
    <div
      className="rates"
    >
      <img
        alt="Star"
        className="star"
        src="stars.svg"
      />
      <h1>
        8.7
      </h1>
    </div>
  </div>
</div>
`;
